File: src/token_manager.py
import httpx
import os
from dotenv import load_dotenv, set_key
from pathlib import Path

load_dotenv()

ENV_PATH = Path(".env")

async def renovar_token():
    print("🔄 Renovando token...")

    async with httpx.AsyncClient() as client:
        response = await client.post(
            "https://api.mercadolibre.com/oauth/token",
            data={
                "grant_type": "refresh_token",
                "client_id": os.getenv("APP_ID"),
                "client_secret": os.getenv("SECRET_KEY"),
                "refresh_token": os.getenv("REFRESH_TOKEN"),
            }
        )
        dados = response.json()

    if "access_token" in dados:
        set_key(ENV_PATH, "ACCESS_TOKEN", dados["access_token"])
        set_key(ENV_PATH, "REFRESH_TOKEN", dados["refresh_token"])
        os.environ["ACCESS_TOKEN"] = dados["access_token"]
        os.environ["REFRESH_TOKEN"] = dados["refresh_token"]
        print("✅ Token renovado com sucesso!")
        return True
    else:
        print(f"❌ Erro ao renovar token: {dados}")
        return False

File: src/api/tokenManager.ts
import { config } from '../core/config';
import { logger } from '../core/logger';
import { salvarEnv } from '../core/envManager';

export async function renovarToken(): Promise<boolean> {
  logger.info('Renovando token ML...');

  try {
    const response = await fetch('https://api.mercadolibre.com/oauth/token', {
      method: 'POST',
      headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
      body: new URLSearchParams({
        grant_type: 'refresh_token',
        client_id: config.APP_ID,
        client_secret: config.SECRET_KEY,
        refresh_token: config.REFRESH_TOKEN,
      }),
    });

    const dados = (await response.json()) as Record<string, unknown>;

    if (typeof dados.access_token === 'string') {
      process.env['ACCESS_TOKEN'] = dados.access_token;
      process.env['REFRESH_TOKEN'] = dados.refresh_token as string;

      salvarEnv({ ACCESS_TOKEN: dados.access_token, REFRESH_TOKEN: dados.refresh_token as string });

      logger.info('Token renovado com sucesso');
      return true;
    } else {
      logger.error({ resposta: dados }, 'Erro ao renovar token ML');
      return false;
    }
  } catch (err) {
    logger.error({ err }, 'Falha na requisição de renovação de token');
    return false;
  }
}
